### PR TITLE
Add search history and persist redux state

### PR DESCRIPTION
**Pull request by simonvadee**
This change is [Reviewable](https://reviewable.io/reviews/arkhn/warehouse-api/133)

### CODE REVIEW

File: front/src/components/search/index.tsx [hunk 1]
```diff
@@ -97,6 +101,8 @@ const Search = (): React.ReactElement => {
     } catch (err) {
       const errMessage = err.response ? err.response.data : err.message;
       setApiErrors([errMessage]);
+      setIsLoading(false);
```
Comment (Jasopaum) [inline front/src/components/search/index.tsx:112]: 👍

File: front/src/redux/reducers.tsx [hunk 2]
```diff
@@ -38,4 +33,18 @@ export const searchParametersReducer = (
...
-export default searchParametersReducer;
+const searchHistoryInitialState: any = [];
+
+export const searchHistoryReducer = (
+  state = searchHistoryInitialState,
+  action: ISimpleAction
+): any => {
+  switch (action.type) {
+    case 'NEW_QUERY': {
+      return [action.payload, ...state];
```
Comment (Jasopaum) [inline front/src/redux/reducers.tsx:44]: Should we only keep a fixed number of queries?

**Review by Jasopaum — COMMENTED** [review]
Looks good, only thing I would have changed is to set a max number of queries in the history

File: front/src/redux/reducers.tsx (hunk 2 above)
Comment (Jasopaum) [reply front/src/redux/reducers.tsx:44]: Also, what about duplicates?

Comment (Jasopaum) [reply front/src/redux/reducers.tsx:44]: When we select the last query and run it again, it will be stored again in redux, right?

Comment (simonvadee) [reply front/src/redux/reducers.tsx:44]: yes I wanted to change that, which is really easy but I guess I was too lazy! will do

**Comment by simonvadee** [comment]
agreed to set a max number of queries in history

File: front/src/redux/reducers.tsx [hunk 3]
```diff
@@ -18,10 +16,7 @@ export const searchParametersReducer = (
     }
 
     case 'ADD_PARAMETER': {
-      state = [
-        ...state,
-        { parameter: '', value: '' },
-      ];
+      state = [...state, { parameter: '', value: '' }];
```
Comment (BPierrick) [inline front/src/redux/reducers.tsx:20]: State should be immutable.

```suggestion
     return [...state, { parameter: '', value: '' }];
```

**Pull request merged**